File: 2021/day_01/lua/sonar_sweep.lua
-- How many measurements are larger than the previous measurement?

package.path = package.path .. ";../../../modules/modules.lua"
local aoc = require("modules")

local test = aoc.fileRead('../input.txt')
-- local test = aoc.fileRead('../test.txt')

local measurement = aoc.extractMesurmentsInTable(test)

local function findHowManyEncreases(measur, line)
    local count = 0
    for i = 2, line do
        if measur[i] > measur[i-1] then
            count = count + 1
        end
    end
    return count
end

local result = findHowManyEncreases(measurement, #measurement)
print("Part one result: " .. result)

-- Part Two

local function measurementTwo(tableMeasure)
    local measure = {}
    for i = 3, #measurement do
        local temp = tableMeasure[i] + tableMeasure[i-1] + tableMeasure[i-2]
        table.insert(measure, temp)
    end
    return measure
end

local partTwoTable = measurementTwo(measurement)

local resultTwo = findHowManyEncreases(partTwoTable, #partTwoTable)
print("Part two result: " .. resultTwo)

File: 2021/day_01/php/sonar_sweep.php
<?php

require "../../modules/modules.php";

// $test = fileRead('../test.txt');
$test = fileRead('../input.txt');
$test = explode("\n", $test);
array_pop($test);

/**
 * howManyIncreases function
 *
 * @param array $arr - Array of measurements
 *
 * @return integer
 */
function howManyIncreases($arr): int
{
    $counter = 0;
    for ($i = 1; $i < count($arr); ++$i) {
        if ($arr[$i] > $arr[$i - 1]) {
            $counter++;
        }
    }
    return $counter;
}

$result = howManyIncreases($test);
echo "Result of part one: " . $result . "\n";

/**
 * tree_measurement function
 *
 * @param array $arr - Array of measurements
 *
 * @return array
 */
function tree_measurement($arr): array
{
    $result = array();
    for ($i = 2; $i < count($arr); ++$i) {
        $temp = $arr[$i] + $arr[$i - 1] + $arr[$i - 2];
        array_push($result, $temp);
    }
    return $result;
}

$partTwoArray = tree_measurement($test);
$partTwo = howManyIncreases($partTwoArray); 
echo "Result of part two: " . $partTwo . "\n";
